docker/bin/zookeeperStart.sh

```diff
@@ -99,9 +98,35 @@ if [[ "$WRITE_CONFIGURATION" == true ]]; then
 
     echo Registering node and writing local configuration to disk.
     java -Dlog4j.configuration=file:"$LOG4J_CONF" -jar /root/zu.jar add $ZKURL $MYID  $ZKCONFIG $DYNCONFIG
-
   fi
 fi
 
+ZOOCFGDIR=/data/conf
+export ZOOCFGDIR
+
+if [ ! -d "$ZOOCFGDIR" ]
+then
+echo "Creating writable conf directory."
+mkdir $ZOOCFGDIR
+cp -f /conf/zoo.cfg $ZOOCFGDIR
+cp -f /conf/log4j.properties $ZOOCFGDIR
+cp -f /conf/log4j-quiet.properties $ZOOCFGDIR
+cp -f /conf/env.sh $ZOOCFGDIR
+fi
+
+if [[ "$WRITE_CONFIGURATION" == false &&  "$REGISTER_NODE" == false ]]; then
+  # We get here only on server restart...
+  echo Printing bootstrap dynamic config file
+  cat $DYNCONFIG
+  echo Printing Static Config
+  STATIC_CONFIG=`cat $ZOOCFGDIR/zoo.cfg`
+  echo Writing configuration gleaned from zookeeper ensemble
+  echo "$CONFIG" | grep -v "^version="> $DYNCONFIG
+  cat $DYNCONFIG
+  #Setting dynamicConfigFile=/data/zoo.cfg.dynamic in zoo.cfg
+  sed -i 's/dynamicConfigFile=.*/dynamicConfigFile=\/data\/zoo\.cfg\.dynamic/g' $ZOOCFGDIR/zoo.cfg
```

Comment:
Let's add a comment here to make the purpose of this replacement clear.  From what I can tell:

```
# This node was found in the zookeeper ensemble on disk, setting dynamic config to the base dynamic config file.
```

Do I understand the purpose of this line correctly?

Reply:
Yup, It sets dynamic configuration file to the current cluster configuration, fetched from zk-client service.